### PR TITLE
Alerting: do not unescape external AM label values

### LINKED ISSUE #44828: Escape regex pattern for label matcher in grafana cloud
If one tries to save the regex pattern `^[a-z0-9-]{1}[a-z0-9-]{0,30}$` right now for grafana cloud it wouldn't work, as the matcher expect the sequence to be escaped by `"`. We need to make sure that before sending the regex to the server we wrap it in `"` so it would become `"^[a-z0-9-]{1}[a-z0-9-]{0,30}$"`.

Screenshot 2022-02-03 at 13 06 58

Steps to reproduce:

- Use grafana cloud
- Create a new notification policy
- Create a matching label policy with the regex pattern mentioned above
- Error will be displayed on save

### PR DESCRIPTION
**Pull request by gillesdemey**
**What this PR does / why we need it**:

Any escaped label values were automatically unescaped, this meant that when any notification policy was being configured with a valid regular expression (like `"^[a-z0-9-]{1}[a-z0-9-]{0,30}$"`) it would be saved properly but decoded on the front-end.

This means that it was not possible to update any existing notification policies or create new ones until the notification policy that caused the issue was either deleted or the label value was updated manually to be wrapped with `"<expression>"`.

**Which issue(s) this PR fixes**:

Fixes #44828

**Special notes for your reviewer**:

If you are using RegEx values you still have to manually wrap them with `"<expression>"`.

I am uncertain if this would break any existing label values but manual testing with various combinations of quotes and escape sequences seem to indicate no breaking changes.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/alertmanager.test.ts [hunk 1]
```diff
@@ -31,17 +31,11 @@ describe('Alertmanager utils', () => {
       });
     });
 
-    it('should parse escaped values correctly', () => {
-      expect(parseMatcher('foo=~"bar\\"baz\\""')).toEqual<Matcher>({
+    it('should not parse escaped values', () => {
```
Comment (gotjosh) [inline public/app/features/alerting/unified/utils/alertmanager.test.ts:34]: I'd document this test to explain _why_ it shouldn't be escaped by using some knowledge from https://github.com/prometheus/alertmanager/blob/4030e3670b359b8814aa8340ea1144f32b1f5ab3/pkg/labels/parse.go#L55-L99

**Comment by achatterjee-grafana** [comment]
Do we need to capture this in the product doc?

**Comment by gillesdemey** [comment]
> Do we need to capture this in the product doc?

I don't think anything needs to be documented on our side; these changes will relay exceptions from Alertmanager to the user and those are pretty straight-forward.

**Review by peterholmberg — APPROVED** [review]
I am happy with this.

**Pull request merged**